session/containerhandler.go

```diff
@@ -39,6 +46,8 @@ type (
 		IsMaster          bool               `json:"isMaster"`
 		ExternalReference *ContainerExternal `json:"externalReference"`
 		Type              string             `json:"type"`
+		// Condition fulfilled  show condition. Value is nil on no show condition, true if it should show and false otherwise
+		Condition *string `json:"condition"`
```

Comment:
this physically hurts🩹

Reply:
> 
> 
> Looks good! I guess docs need an update, so will have another look then.
> 
> What is the reasoning behind picking from visible only for the random mode of the action?

user can only see the visible, there's no tab showing for visualizations not visible, so reason is to chose only among those tabs a user would use.

Don't think docs need an update, maybe some clarification?

Reply:
you mean the tristate, or the fact they've chosen to send a string?

Reply:
should probably make the comment "True" and "False", will update that

Reply:
Maybe throw in a "visible" [here](https://github.com/qlik-oss/gopherciser/blob/80bb4fc10a278d1f8b85431959c11c17ffb1bb60/generatedocs/data/params.json#L225).

I meant why we dont chose from visible in all modes. But it makes sense to leave it up to the script author, when the mode is not random.

Reply:
sure can clarify that part. But as a user I would assume "random" only select a selectable tab.

Reply:
Meant that it is a string in the message.

Reply:
> 
> 
> Maybe throw in a "visible" [here](https://github.com/qlik-oss/gopherciser/blob/80bb4fc10a278d1f8b85431959c11c17ffb1bb60/generatedocs/data/params.json#L225).
> 
done. now I can check the doc checkbox too ;)

Reply:
😆